### PR TITLE
Add ability to listen to group of affix settings

### PR DESCRIPTION
**Pull request by Tim-Brooks**
Currently we have the ability to listen for setting changes to two group
affix settings. However, it is possible that we might have the need to
listen to more than two. This commit adds a method that allows consumer
to listen to a list of affix settings for changes.

### CODE REVIEW

**Comment by tbrooks8** [comment]
https://github.com/elastic/elasticsearch/pull/37678 is the motivation for this change. We could listen to the settings individually, but each change we tear down and reopen connections, so it would be nice if the updates were grouped.

**Review by s1monw — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java [hunk 1]
```diff
@@ -296,6 +296,55 @@ public void apply(Map<String, Tuple<A, B>> values, Settings current, Settings pr
...
+            public boolean hasChanged(Settings current, Settings previous) {
+                return affixUpdaters.stream().anyMatch(au -> au.hasChanged(current, previous));
+            }
+
+            @Override
+            public Map<String, Settings> getValue(Settings current, Settings previous) {
+                Set<String> namespaces = new HashSet<>();
+                Consumer<String> aConsumer = namespaces::add;
+                for (Setting.AffixSetting<?> setting : settings) {
+                    SettingUpdater affixUpdaterA = setting.newAffixUpdater((k, v) -> aConsumer.accept(k), logger, (a, b) ->{});
```
Comment (s1monw) [inline server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java]: do we need the `Consumer<String> aConsumer = namespaces::add;`? I mean we can just call `namespaces.add(k)` instead?

**Comment by danielmitterdorfer** [comment]
@tbrooks8 seems this one never made it to 6.x? Can you please backport it and update the labels here accorgingly?

**Pull request merged**